packages/grid_client/scripts/multiple_vms.ts

```diff
@@ -1,28 +1,34 @@
-import { FilterOptions, MachinesModel } from "../src";
+import { FilterOptions, GridClient, MachinesModel } from "../src";
 import { config, getClient } from "./client_loader";
 import { log } from "./utils";
 
-async function deploy(client, vms) {
+async function deploy(client: GridClient, vms: MachinesModel) {
   const res = await client.machines.deploy(vms);
   log("================= Deploying VM =================");
   log(res);
   log("================= Deploying VM =================");
 }
 
-async function getDeployment(client, vms) {
-  const res = await client.machines.getObj(vms);
+async function getDeployment(client: GridClient, name: string) {
+  const res = await client.machines.getObj(name);
   log("================= Getting deployment information =================");
   log(res);
   log("================= Getting deployment information =================");
 }
 
-async function cancel(client, vms) {
-  const res = await client.machines.delete(vms);
+async function cancel(client: GridClient, name: string) {
+  const res = await client.machines.delete({ name });
   log("================= Canceling the deployment =================");
   log(res);
   log("================= Canceling the deployment =================");
 }
 
+async function getNodeId(client: GridClient, options: FilterOptions) {
+  const nodes = await client.capacity.filterNodes(options);
+
+  return nodes[0]?.nodeId;
+}
```

Comment:
I think you have to ping the node first here before you return it's ID

Reply:
Done.